src/http-api/fetch.ts

```diff
@@ -104,35 +103,29 @@ export class FetchHttpApi<O extends IHttpOpts> {
 
     /**
      * Perform an authorised request to the homeserver.
-     * @param {string} method The HTTP method e.g. "GET".
-     * @param {string} path The HTTP path <b>after</b> the supplied prefix e.g.
+     * @param method - The HTTP method e.g. "GET".
+     * @param path - The HTTP path <b>after</b> the supplied prefix e.g.
      * "/createRoom".
      *
-     * @param {Object=} queryParams A dict of query params (these will NOT be
+     * @param queryParams - A dict of query params (these will NOT be
      * urlencoded). If unspecified, there will be no query params.
      *
-     * @param {Object} [body] The HTTP JSON body.
+     * @param body - The HTTP JSON body.
      *
-     * @param {Object|Number=} opts additional options. If a number is specified,
+     * @param opts - additional options. If a number is specified,
      * this is treated as `opts.localTimeoutMs`.
      *
-     * @param {Number=} opts.localTimeoutMs The maximum amount of time to wait before
-     * timing out the request. If not specified, there is no timeout.
-     *
-     * @param {string=} opts.prefix The full prefix to use e.g.
-     * "/_matrix/client/v2_alpha". If not specified, uses this.opts.prefix.
-     *
-     * @param {string=} opts.baseUrl The alternative base url to use.
-     * If not specified, uses this.opts.baseUrl
-     *
-     * @param {Object=} opts.headers map of additional request headers
-     *
-     * @return {Promise} Resolves to <code>{data: {Object},
-     * headers: {Object}, code: {Number}}</code>.
-     * If <code>onlyData</code> is set, this will resolve to the <code>data</code>
-     * object only.
-     * @return {module:http-api.MatrixError} Rejects with an error if a problem
-     * occurred. This includes network problems and Matrix-specific error JSON.
```

Comment:
Do we really want to get rid of the reject errors? Those aren't implicit by the return type so I think it would be nice to keep this information around

Reply:
This was discussed [today in the Web Chapter Sync](https://docs.google.com/document/d/11-rd31tZxCvtp-8hfGj3sKlAE1m1r9L1n8aklR5ZLqU/edit#bookmark=id.tn0uhbinedyj). The general consensus was that this type of information is valuable to keep around 👍. We agreed that we will try to document what can be thrown from a function, but we won’t claim that we always document everything that is thrown.

Unclear if we should use `@return` or `@throws` 🤷 and I'm not that familiar with JSDoc to know for sure